tests/js/client/server_parameters/auto-refill-index-caches-on-modify-noncluster-off.js

```diff
@@ -0,0 +1,221 @@
+
+const cn = 'UnitTestsCollection';
+const n = 5000;
+
+// wait for metrics to settle. 5 seconds is arbitrary, and may not even
+// be enough on a busy CI server
+const waitForMetrics = () => {
+  require('internal').sleep(5);
+};
+
+function AutoRefillIndexCachesEdge() {
+  'use strict';
+
+  let runCheck = () => {
+    let crsr = db._query(`FOR i IN 0..${n - 1} FOR doc IN ${cn} FILTER doc._from == CONCAT('v/test', i) RETURN doc._from`);
+    let res = crsr.toArray();
+    assertEqual(n, res.length);
+    res.forEach((val, i) => {
+      assertEqual(val, 'v/test' + i);
+    });
+    let stats = crsr.getExtra().stats;
+    assertEqual(0, stats.cacheHits, stats);
+
+    crsr = db._query(`FOR i IN 0..24 FOR doc IN ${cn} FILTER doc._to == CONCAT('v/test', i) RETURN doc._to`);
+    res = crsr.toArray();
+    assertEqual(n, res.length);
+    res.forEach((val, i) => {
+      assertEqual(val, 'v/test' + Math.floor(i / (n / 25)));
+    });
+    stats = crsr.getExtra().stats;
+    assertEqual(0, stats.cacheHits, stats);
+  };
+
+  return {
+    setUp: function() {
+      db._createEdgeCollection(cn);
+    },
+
+    tearDown: function() {
+      db._drop(cn);
+    },
+    
+    testInsertEdge: function() {
+      const oldValue = getMetric("rocksdb_cache_auto_refill_loaded_total");
+      db._query(`FOR i IN 0..${n - 1} INSERT {_from: CONCAT('v/test', i), _to: CONCAT('v/test', (i % 25))} INTO ${cn}`);
+      const newValue = getMetric("rocksdb_cache_auto_refill_loaded_total");
+
+      // 10 here because there may be background operations running that
+      // could affect the metrics
+      assertTrue(newValue - oldValue <= 10, { oldValue, newValue });
+      runCheck();
+    },
+    
+    testUpdateEdge: function() {
+      db._query(`FOR i IN 0..${n - 1} INSERT {_from: CONCAT('v/test', i), _to: CONCAT('v/test', (i % 25))} INTO ${cn}`);
+      const oldValue = getMetric("rocksdb_cache_auto_refill_loaded_total");
+      db._query(`FOR doc IN ${cn} UPDATE doc WITH {value: doc.value + 1} INTO ${cn}`);
+      const newValue = getMetric("rocksdb_cache_auto_refill_loaded_total");
+
+      // 10 here because there may be background operations running that
+      // could affect the metrics
+      assertTrue(newValue - oldValue <= 10, { oldValue, newValue });
+      runCheck();
+    },
+    
+    testReplaceEdge: function() {
+      db._query(`FOR i IN 0..${n - 1} INSERT {_from: CONCAT('v/test', i), _to: CONCAT('v/test', (i % 25))} INTO ${cn}`);
+      const oldValue = getMetric("rocksdb_cache_auto_refill_loaded_total");
+      db._query(`FOR doc IN ${cn} REPLACE doc WITH {_from: doc._from, _to: doc._to, value: doc.value + 1} INTO ${cn}`);
+      const newValue = getMetric("rocksdb_cache_auto_refill_loaded_total");
+
+      // 10 here because there may be background operations running that
+      // could affect the metrics
+      assertTrue(newValue - oldValue <= 10, { oldValue, newValue });
+      runCheck();
+    },
+    
```

Comment:
Need to add tests for remove.
What about tests with `refillIndexCaches: true`?

Reply:
addressed